pkg/controller/podgc/gc_controller_test.go

```diff
@@ -135,90 +151,230 @@ func TestGCTerminated(t *testing.T) {
-				{name: "a", phase: v1.PodFailed},
-				{name: "b", phase: v1.PodSucceeded},
+			name: "nodes present in lister",
+			initialInformerNodes: []*v1.Node{
+				testutil.NewNode("existing1"),
+				testutil.NewNode("existing2"),
+			},
+			delay: 2 * quarantineTime,
+			pods: []*v1.Pod{
+				makePod("a", "existing1", v1.PodRunning),
+				makePod("b", "existing2", v1.PodFailed),
+				makePod("c", "existing2", v1.PodSucceeded),
 			},
-			threshold:       0,
+			itemsInQueue:    0,
+			deletedPodNames: sets.NewString(),
+		},
+		{
+			name: "nodes present in client",
+			initialClientNodes: []*v1.Node{
+				testutil.NewNode("existing1"),
+				testutil.NewNode("existing2"),
+			},
+			delay: 2 * quarantineTime,
+			pods: []*v1.Pod{
+				makePod("a", "existing1", v1.PodRunning),
+				makePod("b", "existing2", v1.PodFailed),
+				makePod("c", "existing2", v1.PodSucceeded),
+			},
+			itemsInQueue:    2,
+			deletedPodNames: sets.NewString(),
+		},
+		{
+			name:  "no nodes",
+			delay: 2 * quarantineTime,
+			pods: []*v1.Pod{
+				makePod("a", "deleted", v1.PodFailed),
+				makePod("b", "deleted", v1.PodSucceeded),
+			},
+			itemsInQueue:    1,
 			deletedPodNames: sets.NewString("a", "b"),
 		},
 		{
-			pods: []nameToPhase{
-				{name: "a", phase: v1.PodRunning},
+			name:  "quarantine not finished",
+			delay: quarantineTime / 2,
+			pods: []*v1.Pod{
+				makePod("a", "deleted", v1.PodFailed),
 			},
-			threshold:       1,
+			itemsInQueue:    0,
+			deletedPodNames: sets.NewString(),
+		},
+		{
+			name:                 "wrong nodes",
+			initialInformerNodes: []*v1.Node{testutil.NewNode("existing")},
+			delay:                2 * quarantineTime,
+			pods: []*v1.Pod{
+				makePod("a", "deleted", v1.PodRunning),
+			},
+			itemsInQueue:    1,
 			deletedPodNames: sets.NewString("a"),
 		},
+		{
+			name:                 "some nodes missing",
+			initialInformerNodes: []*v1.Node{testutil.NewNode("existing")},
+			delay:                2 * quarantineTime,
+			pods: []*v1.Pod{
+				makePod("a", "deleted", v1.PodFailed),
+				makePod("b", "existing", v1.PodFailed),
+				makePod("c", "deleted", v1.PodSucceeded),
+				makePod("d", "deleted", v1.PodRunning),
+			},
+			itemsInQueue:    1,
+			deletedPodNames: sets.NewString("a", "c", "d"),
+		},
+		{
+			name:             "node added to client after quarantine",
+			delay:            2 * quarantineTime,
+			addedClientNodes: []*v1.Node{testutil.NewNode("node")},
+			pods: []*v1.Pod{
+				makePod("a", "node", v1.PodRunning),
+			},
+			itemsInQueue:    1,
+			deletedPodNames: sets.NewString(),
+		},
+		{
+			name:               "node added to informer after quarantine",
+			delay:              2 * quarantineTime,
+			addedInformerNodes: []*v1.Node{testutil.NewNode("node")},
+			pods: []*v1.Pod{
+				makePod("a", "node", v1.PodFailed),
+			},
+			itemsInQueue:    1,
+			deletedPodNames: sets.NewString(),
+		},
+		{
+			name:               "node deleted from client after quarantine",
```

Comment:
Not sure I understand this case - client will never lag comparing to informer.

Reply:
This is sanity check

Reply:
Add a comment then that this is more a sanity-check. I can imagine that at some point this test may start passing and it may be fine to remove it for a good reason.

Reply:
Done